### PR TITLE
Reset the total fit-validation batch progress on epoch

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

The fit-loop validation total counters are supposed to count the total number of batches per epoch. This PR fixes it.

Note that we don't want the same behavior for the validate-loop validation.

This is necessary for #11213 where the total value will be used to update the progress bar.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @carmocca @awaelchli

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

File: tests/loops/test_loops.py [hunk 1]
```diff
@@ -344,7 +344,8 @@ def val_dataloader(self):
     trainer.fit_loop.load_state_dict(checkpoint)
 
     # `nbe_`: non-breaking epoch, as in, no exception will be raised. `be_`: breaking epoch
-    nbe_total_val_batch = stop_epoch * n_dataloaders * n_batches
+    # the fit-validation total batch progress is reset per epoch so it's not counted for the total value.
+    nbe_total_val_batch = 0  # stop_epoch * n_dataloaders * n_batches
```
Comment (Borda) [inline tests/loops/test_loops.py:348]: why to keep this comment?

Comment (carmocca) [reply tests/loops/test_loops.py:348]: because it's the expected `nbe_total_val_batch` value if we didn't do the special behaviour implemented in this PR

**Review by tchaton — APPROVED** [review]
LGTM !
